peps/pep-0784.rst

```diff
@@ -227,31 +226,13 @@ immediately renamed to ``compression._common.streams``. The new name was
 selected due to the current contents of the module being I/O related helpers
 for stream APIs (e.g. ``LZMAFile``) in standard library compression modules.
 
-Compression module migration timeline
--------------------------------------
-
-Existing modules will emit a ``DeprecationWarning`` in the Python
-release following the last Python without the ``compression`` module leaving
-support. For example, if the ``compression`` namespace is introduced in 3.14,
-then the ``DeprecationWarnings`` would be emitted in 3.19, the next release
-after 3.13 reaches end of life. These warnings would begin five years after the
-introduction of ``compression`` namespace. In accordance with :pep:`387`, in
-Python 3.24, five years after the ``DeprecationWarnings`` are added and ten
-years after the new ``compression`` namespace is introduced, the existing
-modules will be removed and code must use the ``compression`` sub-modules. The
-documentation for these modules will be updated to discuss the planned
-deprecation and removal timelines.
-
 
 Backwards Compatibility
 =======================
 
-The main compatibility concern is usage of existing standard library
-compression APIs with the existing import names. These names will be
-deprecated in 3.19 and will be removed in 3.24. Given the long coexistence of
-the modules and a 5 year deprecation period, most users will likely migrate to
-the new import names well before then. Additionally, a libCST codemod can be
-provided to automatically rewrite imports, easing the migration.
+This PEP introduces no backwards incompatible changes. There are currently no
+plans to deprecate or remove the existing compression modules. Any deprecation
+or removal of the existing modules is left to a future decision.
```

Comment:
Is it worth commenting that the expectation of a future PEP is a timeline of at least 5-10 years? Currently it says nothing, which readers may interpret as 'the old names could be at risk at any point'.

Reply:
It could say "... is left to a future decision, but no sooner than 5 years".  Let's see what other SC members think.

Reply:
I almost put a reference to the default deprecation timeline of 5 years but I like the framing of "no sooner than 5 years"

Reply:
"no sooner than" sounds good to me.  The thing to communicate is that there is no breaking change people need to plan for in this PEP and if we ever decide to, it won't be soon.

Reply:
I added this wording in 2ce131663f331ff3fccba08152ab9b104d3b6551.